File: object/object/src/Lec02/RefundCalculator.java
package Lec02;

import Lec02.policy.refund.RefundPolicy;

/**
 * refundCalculate 특 : 환불과 관련된 정보를 갖고 있음
 *                 책임 : 환불 금액 계산
 */
public class RefundCalculator {

    private RefundPolicy refundPolicy;
    private Money fee;

    public Money getFee() {
        return fee;
    }

    public RefundCalculator(RefundPolicy refundPolicy) {
        this.refundPolicy = refundPolicy;
    }


    public Money calculateRefundFee(Screening screening) {
        return refundPolicy.calculateRefundAmount(screening);
    }


}

File: object_Movie/src/main/java_to_kotlin/RefundCalculator.kt
package java_to_kotlin

import java_to_kotlin.policy.refund.RefundPolicy

/**
 * refundCalculate 역할 :
 * 책임 : 환불 금액 계산
 */
class RefundCalculator(private val refundPolicy: RefundPolicy) {
    val fee: Money? = null

    fun calculateRefundFee(refund: Refund): Money? {
        return refundPolicy.calculateRefundAmount(refund)
    }
}
